### PR TITLE
Multiple Issues handled

### PR DESCRIPTION
**Pull request by sdamaraju**
1. Line disappears when in editing mode - 216
2. Lassoing spots - tags - 233
3. Tags moving around the taglist - 236
4. Nullpointer exception handled for setDisplayedSpotsWhileEditing
5. Lassoing spots state doesn't get reset on end draw.
6. deselect the selected spot while creating a new spot.

### CODE REVIEW

**Comment by sdamaraju** [comment]
Issue #216 is fixed as part of #248

File: src/modules/tags/TagsModal.js [hunk 1]
```diff
@@ -43,9 +43,24 @@ const TagsModal = () => {
...
+        selectedSpotsForTagging.map(spot => {
+          if (!useTags.tagSpotExists(tag, spot)) spotsListForTagging.push(spot.properties.id);
+        });
+        let tagCopy = JSON.parse(JSON.stringify(tag));
+        tagCopy.spots = tagCopy.spots.concat(spotsListForTagging);
+        tagsToUpdate.push(tagCopy);
+      });
+      useTags.saveTag(tagsToUpdate);
+    }
     else checkedTagsTemp.map(tag => selectedSpotsForTagging.map(spot => useTags.addTagToSpot(tag, spot)));
```
Comment (jessicagood) [inline src/modules/tags/TagsModal.js:64]: Now there are two `else` statements after the `if`

File: src/modules/tags/useTags.js [hunk 2]
```diff
@@ -158,18 +158,24 @@ const useTags = () => {
...
-  const addTagToSpot = (tag, spot) => {
+  const tagSpotExists = (tag, spot) => {
     if (!tag.spots) tag.spots = [];
-    const spotId = spot ? spot.properties.id : selectedSpot.properties.id;
-    const i = tag.spots.indexOf(spotId);
-    if (i === -1) tag.spots.push(spotId);
-    if (isEmpty(tag.spots)) delete tag.spots;
-    saveTag(tag);
+    const i = tag.spots.indexOf(spot.properties.id);
+    if (i === -1) return false;
```
Comment (jessicagood) [inline src/modules/tags/useTags.js:177]: This can be written more simply `return i !== -1`

File: src/modules/tags/TagsModal.js (hunk 1 above)
Comment (sdamaraju) [reply src/modules/tags/TagsModal.js:64]: .. I don't see the extra else, neither in my code nor in the commit too.

Screen Shot 2021-02-02 at 3 00 42 PM

Comment (jessicagood) [reply src/modules/tags/TagsModal.js:64]: image

Comment (NathanNovak) [reply src/modules/tags/TagsModal.js:64]: I did a merge into the pull request to resolve a conflict this morning.  I didn't know that would mess that up sorry.

Comment (NathanNovak) [reply src/modules/tags/TagsModal.js:64]: Next time there is a conflict I will let you know about it so you can fix it in your branch

**Comment by jessicagood** [comment]
Can you redo the pull request and have each issue solved be its own commit so it's easy to see which code changes solve which issues? This branch is also called SampleDetail which isn't applicable. Make a new branch off master called issue-fixes or something like that and copy in your changes, one issue per commit, noting which issue the commit fixes. That will make it easier to review. Thanks!

**Pull request closed without merging**